### PR TITLE
Type union coercion fails when record types come before others [#258]

### PR DESCRIPTION
**Pull request by dacreify**
# Checklist

- [x] tests
- [x] updated [CHANGELOG](../CHANGELOG.md) (the "unreleased" section)

### CODE REVIEW

File: src/jackdaw/serdes/avro.clj [hunk 1]
```diff
@@ -389,15 +389,17 @@
...
-    (let [[_ unknown-fields _] (clojure.data/diff (set (keys field->schema+coercion))
-                                                  (set (keys clj-map)))]
-      (and (every? (fn [[field-key [^Schema$Field field field-coercion]]]
-                     (let [field-value (get clj-map field-key ::missing)]
-                       (if (= field-value ::missing)
-                         (.defaultValue field)
-                         (match-clj? field-coercion field-value))))
-                   field->schema+coercion)
-           (empty? unknown-fields))))
+    (if (map? clj-map)
```
Comment (cddr) [inline src/jackdaw/serdes/avro.clj:392]: Since the `else` clause returns false, I think this could maybe just be `(when (map? clj-map) ...)`.

**Review by cddr — COMMENTED** [review]
LGTM. Only a minor stylistic suggestion.

File: src/jackdaw/serdes/avro.clj (hunk 1 above)
Comment (dacreify) [reply src/jackdaw/serdes/avro.clj:392]: @cddr That's what I had initially, but then I noticed explicit `(boolean ..` forms in other `match-clj?` implementations, and so thought maybe returning falsey `nil` was against the contract. If not glad to change!

Comment (dacreify) [reply src/jackdaw/serdes/avro.clj]: Made this change and also updated `CHANGELOG`. Tried to re-request review but the button isn't doing anything. 🤷🏼‍♂️

**Pull request closed without merging**